examples/delegated-fungible/src/contract.rs

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+#![cfg_attr(target_arch = "wasm32", no_main)]
+
+mod state;
+
+use delegated_fungible::{
+    Account, DelegatedFungibleTokenAbi, InitialState, Message, Operation, Parameters,
+};
+use linera_sdk::{
+    linera_base_types::{AccountOwner, Amount, WithContractAbi},
+    views::{RootView, View},
+    Contract, ContractRuntime,
+};
+
+use self::state::DelegatedFungibleTokenState;
+
+pub struct DelegatedFungibleTokenContract {
+    state: DelegatedFungibleTokenState,
+    runtime: ContractRuntime<Self>,
+}
+
+linera_sdk::contract!(DelegatedFungibleTokenContract);
+
+impl WithContractAbi for DelegatedFungibleTokenContract {
+    type Abi = DelegatedFungibleTokenAbi;
+}
+
+impl Contract for DelegatedFungibleTokenContract {
+    type Message = Message;
+    type Parameters = Parameters;
+    type InstantiationArgument = InitialState;
+    type EventValue = ();
+
+    async fn load(runtime: ContractRuntime<Self>) -> Self {
+        let state = DelegatedFungibleTokenState::load(runtime.root_view_storage_context())
+            .await
+            .expect("Failed to load state");
+        DelegatedFungibleTokenContract { state, runtime }
+    }
+
+    async fn instantiate(&mut self, state: Self::InstantiationArgument) {
+        // Validate that the application parameters were configured correctly.
+        let _ = self.runtime.application_parameters();
+
+        let mut total_supply = Amount::ZERO;
+        for value in state.accounts.values() {
+            total_supply.saturating_add_assign(*value);
+        }
+        if total_supply == Amount::ZERO {
+            panic!("The total supply is zero, therefore we cannot instantiate the contract");
+        }
+        self.state.initialize_accounts(state).await;
+    }
+
+    async fn execute_operation(&mut self, operation: Self::Operation) -> Self::Response {
+        match operation {
+            Operation::Approve {
+                owner,
+                spender,
+                allowance,
+            } => {
+                self.runtime
+                    .check_account_permission(owner)
+                    .expect("Permission for Transfer operation");
```

Comment:
`check_account_permission` returns true if `owner` is equivalent to either `authenticated_caller_id` or `authenticated_signer`. I'm wondering:
1. does this work for all combinations of callers:
  - Solidity contract, 
  - Wasm contract, 
  - EVM account
  - Linera account
2. Do we even need to call it manually? in EVM world, the `owner` of the `Approve` call will be `msg.sender` (direct caller). Maybe we can pull that information from the runtime rather than requiring in the operation? I.e. it should be `authenticated_caller_id` (and for direct calls, that should be equivalent to `authenticated_signer`?)